### PR TITLE
PYTHON-3066 Test against Apple silicon in Evergreen

### PR DESCRIPTION
**Pull request by juliusgeo**

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -2639,6 +2648,23 @@ buildvariants:
     - ".4.0"
     - ".3.6"
 
+- matrix_name: "test-macos-arm64"
+  matrix_spec:
+    platform:
+      - macos-1100-arm64
+    auth: "*"
+    ssl: "*"
```
Comment (ShaneHarvey) [inline .evergreen/config.yml:2656]: Let's use the `auth-ssl: "*"` axis instead. That way we can test auth/ssl noauth/nossl without needing the exclude_spec.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: .evergreen/config.yml (hunk 1 above)
Comment (juliusgeo) [reply .evergreen/config.yml]: Done.

File: .evergreen/config.yml [hunk 2]
```diff
@@ -2639,6 +2648,18 @@ buildvariants:
     - ".4.0"
     - ".3.6"
 
+- matrix_name: "test-macos-arm64"
+  matrix_spec:
+    platform:
+      - macos-1100-arm64
+    auth-ssl: "*"
+  display_name: "${platform} ${auth} ${ssl}"
```
Comment (ShaneHarvey) [inline .evergreen/config.yml:2656]: `${auth} ${ssl}` -> `${auth-ssl}`

Comment (juliusgeo) [reply .evergreen/config.yml]: Fixed.

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**